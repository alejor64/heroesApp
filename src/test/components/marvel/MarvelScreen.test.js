import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import MarvelScreen from '../../../components/marvel/MarvelScreen'

describe('Pruebas en el componente <DcScreen />', () => {
    test('Debe renderizar el componente', () => {
        const wrapper = mount(
            <MemoryRouter>
                <MarvelScreen />
            </MemoryRouter>
        )
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('HeroesList').exists()).toBe(true)
        expect(wrapper.find('h1').text()).toBe('Marvel')
    })
    
})
