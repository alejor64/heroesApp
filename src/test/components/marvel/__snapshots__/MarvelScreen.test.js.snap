// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <DcScreen /> Debe renderizar el componente 1`] = `
<div>
  <h1>
    Marvel
  </h1>
  <hr />
  <div
    className="card-columns animate__animated animate__fadeIn"
  >
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Spider Man"
            className="card-img"
            src="./assets/heroes/marvel-spider.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Spider Man
            </h5>
            <p
              className="card-text"
            >
              Peter Parker
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Amazing Fantasy #15
              </small>
            </p>
            <a
              href="/heroe/marvel-spider"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Captain America"
            className="card-img"
            src="./assets/heroes/marvel-captain.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Captain America
            </h5>
            <p
              className="card-text"
            >
              Steve Rogers
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Captain America Comics #1
              </small>
            </p>
            <a
              href="/heroe/marvel-captain"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Iron Man"
            className="card-img"
            src="./assets/heroes/marvel-iron.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Iron Man
            </h5>
            <p
              className="card-text"
            >
              Tony Stark
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Tales of Suspense #39
              </small>
            </p>
            <a
              href="/heroe/marvel-iron"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Thor"
            className="card-img"
            src="./assets/heroes/marvel-thor.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Thor
            </h5>
            <p
              className="card-text"
            >
              Thor Odinson
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Journey into Myster #83
              </small>
            </p>
            <a
              href="/heroe/marvel-thor"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Hulk"
            className="card-img"
            src="./assets/heroes/marvel-hulk.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hulk
            </h5>
            <p
              className="card-text"
            >
              Bruce Banner
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                The Incredible Hulk #1
              </small>
            </p>
            <a
              href="/heroe/marvel-hulk"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Wolverine"
            className="card-img"
            src="./assets/heroes/marvel-wolverine.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Wolverine
            </h5>
            <p
              className="card-text"
            >
              James Howlett
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                The Incredible Hulk #180
              </small>
            </p>
            <a
              href="/heroe/marvel-wolverine"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Daredevil"
            className="card-img"
            src="./assets/heroes/marvel-daredevil.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Daredevil
            </h5>
            <p
              className="card-text"
            >
              Matthew Michael Murdock
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Daredevil #1
              </small>
            </p>
            <a
              href="/heroe/marvel-daredevil"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Hawkeye"
            className="card-img"
            src="./assets/heroes/marvel-hawkeye.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Hawkeye
            </h5>
            <p
              className="card-text"
            >
              Clinton Francis Barton
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Tales of Suspense #57
              </small>
            </p>
            <a
              href="/heroe/marvel-hawkeye"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Cyclops"
            className="card-img"
            src="./assets/heroes/marvel-cyclops.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Cyclops
            </h5>
            <p
              className="card-text"
            >
              Scott Summers
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                X-Men #1
              </small>
            </p>
            <a
              href="/heroe/marvel-cyclops"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="Silver Surfer"
            className="card-img"
            src="./assets/heroes/marvel-silver.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Silver Surfer
            </h5>
            <p
              className="card-text"
            >
              Norrin Radd
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                The Fantastic Four #48
              </small>
            </p>
            <a
              href="/heroe/marvel-silver"
              onClick={[Function]}
            >
              More about...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
